pkg/k8s/node.go

```diff
@@ -56,17 +62,33 @@ func ParseNode(k8sNode *slim_corev1.Node, source source.Source) *nodeTypes.Node
 		if addr.Address == "" {
 			continue
 		}
+		addrGroup := nodeAddressGroup{
+			typ: addr.Type,
+		}
 		ip := net.ParseIP(addr.Address)
-		if ip == nil {
+		switch {
+		case ip != nil && ip.To4() != nil:
+			addrGroup.family = slim_corev1.IPv4Protocol
+		case ip != nil && ip.To16() != nil:
+			addrGroup.family = slim_corev1.IPv6Protocol
+		default:
 			scopedLog.WithFields(logrus.Fields{
 				logfields.IPAddr: addr.Address,
-				"type":           addr.Type,
+				logfields.Type:   addr.Type,
 			}).Warn("Ignoring invalid node IP")
 			continue
 		}
+		_, groupFound := addrGroups[addrGroup]
+		if groupFound && len(k8sNode.Spec.PodCIDRs) != 2 {
```

Comment:
This logic doesn't seem sufficiently precise to me. Could it be the case that we have exactly two v4 CIDRs and we should still print the warning?

Reply:
@asauber thanks for the review. I have updated `ParseNode()` to now begin the next iteration of node addresses if the address is invalid. A node can have at most two "InternalIP" and "ExternalIP" addresses as long as each is a separate address family, e.g. IPv4 or IPv6. See https://github.com/cilium/cilium/pull/20811#issuecomment-1235122629 for additional context. I have also removed `len(k8sNode.Spec.PodCIDRs) != 2` from this LOC since it's being checked later in `ParseNode()` and is not applicable to the management of node status addresses.